enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -2577,17 +2580,24 @@ func (e *partitionEvictor) generateSamplesForEviction(quitChan chan struct{}) er
 		shouldSleep := e.sizeBytes <= int64(SamplerSleepThreshold*float64(e.part.MaxSizeBytes))
 		e.mu.Unlock()
 		if shouldSleep {
+			samplerDelay := time.NewTimer(SamplerSleepDuration)
 			select {
 			case <-quitChan:
 				return nil
 			case <-samplerDelay.C:
 			}
+			timeutil.StopAndDrainTimer(samplerDelay)
```

Comment:
We should add some documentation to this function. Draining the timer is only necessary if you plan to reuse it or if there's a chance the timer would frequently not run to completion. Neither is true here. 
Here a simple time.After() would suffice. Same below.

Reply:
I changed to use "After" here. But I needed the timer below to avoid memory leak when case e.samples <- sample is run.